### ISSUE TITLE
Escaping property values

### ISSUE THREAD

**Issue by DudarevDaniel** [polagoab/merge-properties-maven-plugin#4]
Hello,

I'm trying to use your plugin to merge two property files. All is fine, but one thing:
plugin escapes all ":" and "=" symbols in property values. For example I have property:
   `hibernate.connection.url=jdbc:mysql:localhost...`
And after merge I have:
  `hibernate.connection.url=jdbc\:mysql\:localhost...`
Is there any way to turn off such escaping? I tried to set:

```
<escapeWindowsPaths>false</escapeWindowsPaths>
<escapeString>:</escapeString>
```

But it doesn't work. I need to turn off escaping of ":" and "=" symbols.

**Comment by perlan** [comment]
Hi,

The plugin uses a standard `java.util.Properties` class for merging all properties. The escape of ':' and '=', among others, is performed by the `Properties.store()` method and by looking at the source there seems to be no way to tell the that method to skip the escaping part. In fact, the escaping of ':' and '=' is described in the javadoc for [`Properties.load()`](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html#load-java.io.Reader-).

Now, I agree that the escaping in your example is unnecessary but it is not an easy fix unless the plugin implements it's own serializer for the java properties file.

**Comment by hanagiat** [comment]
I have the same problem. The workaround I am using is to let the plugin do the merge, and re-write the properties file unescaped.
To do so, I create another plugin that uses the mojo executor library (https://github.com/TimMoore/mojo-executor).

>         executeMojo(
>                 plugin(
>                         groupId("org.polago.maven.plugins"),
>                         artifactId("merge-properties-maven-plugin"),
>                         version("1.1")
>                 ),
>                 goal("merge"),
>                 configuration(
>                         element(name("overwriteProperties"), "true"),
>                         element(name("outputDirectory"), outputDir.getAbsolutePath()),
>                         element(name("outputFile"), "out.properties"),
>                         element(name("resources"), element("resource",
>                                 element("directory", "src/main/resources"),
>                                 element("includes", element("include", "application.properties"))),
>                                 element("resource",
>                                         element("directory", "src/main/resources"),
>                                         element("includes", element("include", "application-2.properties"))))
>                 ),
>                 executionEnvironment(
>                         mavenProject,
>                         mavenSession,
>                         pluginManager
>                 )
>         );
>         try {
>             unescapePropertiesFile(new File(outputDir.getAbsolutePath(), "out.properties"));
>         } catch (IOException e) {
>             throw new MojoExecutionException("Failed to unescape file", e);
>         }
> 

unescapePropertiesFile method reads the output file by Properties class so the values are unescaped, then serialize them as plain text lines:

>     private static void unescapePropertiesFile(File propertiesFile) throws IOException {
>         Properties props = new Properties();
>         props.load(new FileInputStream(propertiesFile));
> 
>         String fileName = propertiesFile.getName();
>         File fout = new File(propertiesFile.getParentFile(), fileName +"temp");
>         FileOutputStream fos = new FileOutputStream(fout);
> 
>         BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(fos));
> 
>         Iterator<Map.Entry<Object, Object>> iterator = props.entrySet().iterator();
>         while (iterator.hasNext()){
>             Map.Entry<Object, Object> next = iterator.next();
>             String line = String.format("%s=%s", next.getKey(), next.getValue());
>             bw.write(line);
>             bw.newLine();
>         }
>         bw.close();
> 
>         propertiesFile.delete();
>         fout.renameTo(propertiesFile);
>     }
> 
We can extend the merge-properties-maven-plugin to have an "unescape" flag, that runs the method unescapePropertiesFile() after output file generation

**Comment by perlan** [comment]
You seems to have find a creative workaround for the limitations of the Properties serialization. It actually got me thinking that maybe we could provide a custom InputStream that performs the unescaping before the file is saved in the first place. Now, the real question is if this approach is better than implementing a custom serializer that also could sort the keys.

**Comment by perlan** [comment]
This issue is fixed as a side effect of fixing #9 due to switching from using java.util.Properties to commons-configuration for serializing the merged properties file
